src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/specifications/PersonSpecification.kt

```diff
@@ -67,35 +66,46 @@ object PersonSpecification {
     }
   }
 
-  fun levenshteinPostcodes(postcodes: Set<String>, limit: Int = 1): Specification<PersonEntity> {
+  fun exactMatchPostcodePrefix(postcodes: Set<String>, prefixLength: Int = 3): Specification<PersonEntity> {
```

Comment:
Will this match the first 3 characters of a postcode?
It is a bit arbitrary to do that and will result some matches being 10x larger in area than others.
e.g. SW1 will match SW1, SW10, SW11 etc whereas M33 will match only one area
I know we haven't yet done any normalisation here and are mainly looking for speed, but this will make some matches look better than they actually are.

Reply:
Yeah, potential here for it to return more candidates here. But, checked this with robin seems to think its a sensible solution for the mean time until we get the proper blocking rules